R/helper_additional_functions.R

```diff
@@ -213,7 +218,33 @@ clean_pbp <- function(pbp, ...) {
         dplyr::vars(.data$passer_id, .data$rusher_id, .data$receiver_id, .data$id, ends_with("player_id")),
         update_ids, legacy_id_map) %>%
       dplyr::arrange(.data$index) %>%
-      dplyr::select(-"index")
+      dplyr::select(-"index") %>%
+      # add action player
+      dplyr::mutate(
+        fantasy_player_name = case_when(
+          !is.na(.data$rusher_player_name) ~ .data$rusher_player_name,
+          is.na(.data$rusher_player_name) & !is.na(.data$receiver_player_name) ~ .data$receiver_player_name,
+          TRUE ~ NA_character_
+        ),
+        fantasy_player_id = case_when(
+          !is.na(.data$rusher_player_id) ~ .data$rusher_player_id,
+          is.na(.data$rusher_player_id) & !is.na(.data$receiver_player_id) ~ .data$receiver_player_id,
+          TRUE ~ NA_character_
+        ),
+        fantasy = case_when(
+          !is.na(.data$rusher) ~ .data$rusher,
+          is.na(.data$rusher) & !is.na(.data$receiver) ~ .data$receiver,
+          TRUE ~ NA_character_
```

Comment:
See my comment above: the current implementation means if the QB scrambles `fantasy_player_name` will be the QB's name whereas `fantasy` will be `NA`

Reply:
I guess we should put `TRUE ~ passer` in this one. But thinking about it more I'm not sure how useful this version of the stat is

Reply:
What about something like

```
        fantasy = case_when(
          !is.na(.data$rusher) ~ .data$rusher,
          is.na(.data$rusher) & !is.na(.data$receiver) ~ .data$receiver,
         .data$incomplete_pass == 0 & .data$interception == 0 & .data$qb_sack ==0 ~ .data$passer,
          TRUE ~ NA_character_
```

This would catch scrambles but not QB sacks or throwaways which would make it similar to the other one

Reply:
I think that's a good idea. But there is a bug. The variable is called `sack`

```r
fantasy = case_when(
      !is.na(.data$rusher) ~ .data$rusher,
      is.na(.data$rusher) & !is.na(.data$receiver) ~ .data$receiver,
      .data$incomplete_pass == 0 & .data$interception == 0 & .data$sack == 0 ~ .data$passer,
      TRUE ~ NA_character_
    )
```

Reply:
``` r
library(dplyr, warn.conflicts = FALSE)
nflfastR::load_pbp(2020, qs = TRUE) %>%
  dplyr::filter(qb_scramble == 1) %>%
  dplyr::mutate(
    fantasy_player_name = case_when(
      !is.na(.data$rusher_player_name) ~ .data$rusher_player_name,
      is.na(.data$rusher_player_name) & !is.na(.data$receiver_player_name) ~ .data$receiver_player_name,
      TRUE ~ NA_character_
    ),
    fantasy = case_when(
      !is.na(.data$rusher) ~ .data$rusher,
      is.na(.data$rusher) & !is.na(.data$receiver) ~ .data$receiver,
      .data$incomplete_pass == 0 & .data$interception == 0 & .data$sack == 0 ~ .data$passer,
      TRUE ~ NA_character_
    )
  ) %>%
  dplyr::select(
    passer_player_name,
    rusher_player_name,
    fantasy_player_name,
    passer,
    rusher,
    fantasy
  ) %>% 
  head(10)
#> # A tibble: 10 x 6
#>    passer_player_na~ rusher_player_na~ fantasy_player_n~ passer  rusher fantasy 
#>    <chr>             <chr>             <chr>             <chr>   <chr>  <chr>   
#>  1 <NA>              K.Murray          K.Murray          K.Murr~ <NA>   K.Murray
#>  2 <NA>              K.Murray          K.Murray          K.Murr~ <NA>   K.Murray
#>  3 <NA>              K.Murray          K.Murray          K.Murr~ <NA>   K.Murray
#>  4 <NA>              K.Murray          K.Murray          K.Murr~ <NA>   K.Murray
#>  5 <NA>              K.Murray          K.Murray          K.Murr~ <NA>   K.Murray
#>  6 <NA>              K.Murray          K.Murray          K.Murr~ <NA>   K.Murray
#>  7 <NA>              K.Murray          K.Murray          K.Murr~ <NA>   K.Murray
#>  8 <NA>              K.Murray          K.Murray          K.Murr~ <NA>   K.Murray
#>  9 <NA>              J.Garoppolo       J.Garoppolo       J.Garo~ <NA>   J.Garop~
#> 10 <NA>              M.Stafford        M.Stafford        M.Staf~ <NA>   M.Staff~
```

<sup>Created on 2021-02-12 by the [reprex package](https://reprex.tidyverse.org) (v1.0.0)</sup>

Reply:
lol I forgot we had a `qb_scramble` column. Should I use that instead?

Reply:
I am not sure tbh. Is it the same filter? I am not used to those columns.

Reply:
Yes `pass_finder` and the creation of `qb_scramble` both search play description for scramble. So I'm using `qb_scramble` to make it more obvious what we're doing